### PR TITLE
add ETag to aggregator group list

### PR DESCRIPTION
**Pull request by alexzielenski**
refactors shared code into etag.go

based on #14

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/endpoints/discovery/version.go [hunk 1]
```diff
@@ -18,49 +18,59 @@ package discovery
...
 // APIVersionHandler creates a webservice serving the supported resources for the version
 // E.g., such a web service will be registered at /apis/extensions/v1beta1.
 type APIVersionHandler struct {
 	serializer runtime.NegotiatedSerializer
 
-	groupVersion      schema.GroupVersion
-	apiResourceLister APIResourceLister
+	groupVersion schema.GroupVersion
+	resourceList *metav1.APIResourceList
+	hash         string
```
Comment (apelisse) [inline staging/src/k8s.io/apiserver/pkg/endpoints/discovery/version.go:38]: These two things feel related and probably should be updated together, can we force the code to instantiate them together?

Comment (apelisse) [reply staging/src/k8s.io/apiserver/pkg/endpoints/discovery/version.go:38]: Forget it, this is super small, that's fine.

File: staging/src/k8s.io/apiserver/pkg/endpoints/discovery/etag.go [hunk 2]
```diff
@@ -21,14 +21,97 @@ import (
...
 // used in discovery endpoint responses
 
-func CalculateETag(resurces *metav1.APIResourceList) (string, error) {
+// Unsure if there is an existing interface for this
+type Marshalable interface {
+	Marshal() ([]byte, error)
+}
+
+// Attaches Cache-Busting functionality to an endpoint
+// 	* Immutability Response Header
```
Comment (apelisse) [inline staging/src/k8s.io/apiserver/pkg/endpoints/discovery/etag.go]: nit, this one isn't indented properly?

**Comment by apelisse** [comment]
Looks good to me

File: staging/src/k8s.io/apiserver/pkg/endpoints/discovery/etag.go (hunk 2 above)
Comment (alexzielenski) [reply staging/src/k8s.io/apiserver/pkg/endpoints/discovery/etag.go]: tab character snuck in somehow

**Pull request merged**